src/applications/appeals/995/utils/contactInfo.js

```diff
@@ -53,3 +55,21 @@ export const hasMobilePhone = formData =>
 
 export const hasHomeAndMobilePhone = formData =>
   hasHomePhone(formData) && hasMobilePhone(formData);
+
+/**
+ * Set sessionStorage of last edited contact field. We could have used
+ * selectMostRecentlyUpdatedField from the VAP service instead of using
+ * sessionStorage, but we wouldn't know if the edit was canceled
+ * @param {string} key - ID of contact info input
+ * @param {string} state - either "updated" or "canceled"
+ */
+export const setReturnState = (key = '', state = '') =>
+  window.sessionStorage.setItem(CONTACT_EDIT, `${key},${state}`);
+/**
+ * Get ID and state of last edited contact field so we know where to move focus
+ * @returns {Array} Array with ID at index zero, and state at index one
+ */
+export const getReturnState = () =>
+  window.sessionStorage.getItem(CONTACT_EDIT) || '';
+export const clearReturnState = () =>
+  window.sessionStorage.removeItem(CONTACT_EDIT);
```

Comment:
This is cool! I think this would make a good reusable hook if it was refactored to maybe use serialized JSON data instead or some more generic implementation.

I was looking at this hook library example

https://github.com/streamich/react-use/blob/master/src/useSessionStorage.ts

but also noticed that check-in has their own hook for using session storage as well.

https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/check-in/hooks/useSessionStorage.jsx

Something to noodle over, and maybe add to platform code in the future.

Reply:
That's a good idea!